libs/server/Servers/GarnetServerOptions.cs

```diff
@@ -246,6 +246,11 @@ public class GarnetServerOptions : ServerOptions
         /// </summary>
         public int ThreadPoolMaxThreads = 0;
 
+        /// <summary>
+        /// Maximum client connection limit
+        /// </summary>
+        public int ConnectionLimit = 0;
```

Comment:
We should let `0` denote "no connections allowed" which might be useful if for some reason people want to run the server without allowing any network connections. `-1` or `int.MaxValue` can be the default to allow unlimited connections.